Zend/zend_compile.c

```diff
@@ -3044,24 +3044,26 @@ uint32_t zend_compile_args(zend_ast *ast, zend_function *fbc) /* {{{ */
 ZEND_API zend_uchar zend_get_call_op(const zend_op *init_op, zend_function *fbc) /* {{{ */
 {
 	if (fbc) {
-		if (fbc->type == ZEND_INTERNAL_FUNCTION) {
+		if (fbc->type == ZEND_INTERNAL_FUNCTION && !(CG(compiler_options) & ZEND_COMPILE_IGNORE_INTERNAL_FUNCTIONS)) {
 			if (init_op->opcode == ZEND_INIT_FCALL && !zend_execute_internal) {
 				if (!(fbc->common.fn_flags & (ZEND_ACC_ABSTRACT|ZEND_ACC_DEPRECATED|ZEND_ACC_HAS_TYPE_HINTS|ZEND_ACC_RETURN_REFERENCE))) {
 					return ZEND_DO_ICALL;
 				} else {
 					return ZEND_DO_FCALL_BY_NAME;
 				}
 			}
-		} else {
+		} else if (!(CG(compiler_options) & ZEND_COMPILE_IGNORE_USER_FUNCTIONS)){
 			if (zend_execute_ex == execute_ex && !(fbc->common.fn_flags & ZEND_ACC_ABSTRACT)) {
 				return ZEND_DO_UCALL;
 			}
 		}
-	} else if (zend_execute_ex == execute_ex &&
-	           !zend_execute_internal &&
-	           (init_op->opcode == ZEND_INIT_FCALL_BY_NAME ||
-	            init_op->opcode == ZEND_INIT_NS_FCALL_BY_NAME)) {
-		return ZEND_DO_FCALL_BY_NAME;
+	} else if (!(CG(compiler_options) & (ZEND_COMPILE_IGNORE_USER_FUNCTIONS|ZEND_COMPILE_IGNORE_INTERNAL_FUNCTIONS))) {
```

Comment:
Why this check? DO_FCALL_BY_NAME only depends on it being a function (as opposed to a method) and having no executor hooks. It does not use any knowledge about specific functions.

Reply:
It seems nicer from a tooling perspective to get one set of call opcodes when you have set those flags, but I suppose it's not the end of the world to preserve that behaviour ...